articles/building-an-object-detection-application-with-tensorfow.js-and-react.js/index.md

```diff
@@ -0,0 +1,153 @@
+
+### Accessing the React Computer Vision Template
+
+First, we are going to clone my publicly available [ReactComputerVisionTemplate](https://github.com/wmkinyan/ReactComputerVisionTemplate) on Github. This template contains all the code we need to get started.
+
+In our command-line interface, go to the drive that you want to clone into. We will need to run the following command to clone the template:
+
+```bash
+git clone https://github.com/wmkinyan/ReactComputerVisionTemplate
+```
+
+Please make sure the 'git' command is installed before you try the above command. If not, you can install 'git' by issuing the command:
+
+```bash
+sudo apt install git
+```
+
+Secondly, we will need to open the template on VSCode by issuing the following command on the command line interface:
+
+```bash
+code .
+```
+
+That's 'code' followed with a full stop. That command should fire up our code editor.
+
+The next step that we need to do is start our React app using the 'npm start' command. This command is going to start our react app and open a new browser. It will go directly to the 'localhost 3000' by default. This is where our React app is going to start. 
+
+Please make sure the 'npm' command is installed on your computer before issuing the 'npm start' command. If not, install it using the following command:
+
+```bash
+sudo apt install npm
+```
+After issuing the 'npm install' command, you should see a 'node_modules' folder installed in your list of files in the ReactComputerVisionTemplate folder.
+
+### Installing TensorFow.js and setting up pre-built models
+
+We will use Microsoft's [COCO SSD](https://github.com/tensorflow/tfjs-models/tree/master/coco-ssd) pre-trained model, which allows us to perform real-time object detection on images.
+
+We will install TensorFlow and the pre-built model into our application by issuing the following commands:
+
+```bash
+    import * as tf from "@tensorflow/tfjs";
+    import * as cocossd from "@tensorflow-models/coco-ssd";
+```
+
+The first command will first import the TensorFlow module into our application. The second command then imports the COCO SSD model from the TensorFlow model.
+
+### Building the React.js app
+
+We first import all the dependencies on the 'App.js' folder, including the React.js library, TensorFlow.js, and the COCO SSD pre-trained model. It imports the required dependencies into our application. 
+
+```bash
+import React, { useRef, useState, useEffect } from "react";
+import * as tf from "@tensorflow/tfjs";
+import * as cocossd from "@tensorflow-models/coco-ssd";
+import Webcam from "react-webcam";
+import "./App.css";
+import { drawRect } from "./utilities";
+```
+
+The next thing that we are going to do is to load our network into our model. This is achieved by issuing the following command:
+
+```bash
+    const net = await cocossd.load();
+```
+
+We first create a new variable called 'net'. Because our function is asynchronous, we wait for the 'cocossd' model to load.
```

Comment:
What would happen if our function would be of a synchronous paradigm? What is the significance of the asynchronous function in this context?